pilot_experiment/functions/process_data.py

```diff
@@ -0,0 +1,179 @@
+'''
+Author: Alexandra Lee
+Date Created: 17 April 2020
+
+Scripts to process expression data for pilot experiment
+'''
+import pandas as pd
+import os
+import random
+
+import warnings
+
+def fxn():
+    warnings.warn("deprecated", DeprecationWarning)
+
+with warnings.catch_warnings():
+    warnings.simplefilter("ignore")
+    fxn()
+
+def select_expression_data(input_data_file,
+metadata_file,
+lst_experiments,
+output_file):
+    """
+    Returns a subset of expression data, specified by the experiment_ids
+    provided by the user, to use for downstream exploration analysis
+
+    Arguments
+    ----------
+    input_data_file: str
+        File containing normalized gene expression data
+
+        ------------------------------| PA0001 | PA0002 |...
+        05_PA14000-4-2_5-10-07_S2.CEL | 0.8533 | 0.7252 |...
+        54375-4-05.CEL                | 0.7789 | 0.7678 |...
+        ...                           | ...    | ...    |...
+    
+    metadata_file: str
+        File containing the mapping between experiment ids and sample ids. This 
+        mapping can be used to link sample ids in the input_data_file with the 
+        experiment ids provided by the user
+
+    lst_experiments: list
+        List of experiment ids
+
+    output_file: str
+        Filename to save selected expression data
+
+    """
+    # Read data
+    normalized_data = pd.read_table(
+        input_data_file,
+        header=0,
+        sep='\t',
+        index_col=0).T
+
+    # Read metadata
+    metadata = pd.read_table(
+        metadata_file,
+        header=0,
+        sep='\t',
+        index_col=0)
+        
+    map_experiment_sample = metadata[['sample_name', 'ml_data_source']]
+    
+    # Get expression data associated with experiment_id
+    selected_mapping = map_experiment_sample.loc[lst_experiments]
+    selected_sample_ids = list(selected_mapping['ml_data_source'].values)
+
+    selected_data = normalized_data.loc[selected_sample_ids]
+
+    print(selected_data.shape)
```

Comment:
Good point. I'm having it print just as a sanity check since I was saving the data to a file, so that I could run the script just once.  I can remove the print statement